app/src/main/java/com/keylesspalace/tusky/util/MediaUtils.java

```diff
@@ -176,4 +180,81 @@ public static int calculateInSampleSize(BitmapFactory.Options options, int reqWi
 
         return inSampleSize;
     }
+
+    @Nullable
+    public static Bitmap reorientBitmap(Bitmap bitmap, int orientation) {
+        Matrix matrix = new Matrix();
+        switch (orientation) {
+            default:
+            case ExifInterface.ORIENTATION_NORMAL: {
+                return bitmap;
+            }
+            case ExifInterface.ORIENTATION_FLIP_HORIZONTAL: {
+                matrix.setScale(-1, 1);
+                break;
+            }
+            case ExifInterface.ORIENTATION_ROTATE_180: {
+                matrix.setRotate(180);
+                break;
+            }
+            case ExifInterface.ORIENTATION_FLIP_VERTICAL: {
+                matrix.setRotate(180);
+                matrix.postScale(-1, 1);
+                break;
+            }
+            case ExifInterface.ORIENTATION_TRANSPOSE: {
+                matrix.setRotate(90);
+                matrix.postScale(-1, 1);
+                break;
+            }
+            case ExifInterface.ORIENTATION_ROTATE_90: {
+                matrix.setRotate(90);
+                break;
+            }
+            case ExifInterface.ORIENTATION_TRANSVERSE: {
+                matrix.setRotate(-90);
+                matrix.postScale(-1, 1);
+                break;
+            }
+            case ExifInterface.ORIENTATION_ROTATE_270: {
+                matrix.setRotate(-90);
+                break;
+            }
+        }
+        try {
+            Bitmap result = Bitmap.createBitmap(bitmap, 0, 0, bitmap.getWidth(),
+                    bitmap.getHeight(), matrix, true);
+            if (!bitmap.sameAs(result)) {
+                bitmap.recycle();
+            }
+            return result;
+        } catch (OutOfMemoryError e) {
+            return null;
```

Comment:
would giving the user an error just cause another OOM at this point?

Reply:
No, garbage collector frees up the memory reserved for the bitmap